HLTrigger/Muon/plugins/HLTMuonTrkFilter.cc

```diff
@@ -27,24 +27,27 @@
 
 #include "DataFormats/Math/interface/deltaR.h"
 
-HLTMuonTrkFilter::HLTMuonTrkFilter(const edm::ParameterSet& iConfig) : HLTFilter(iConfig) {
-  m_muonsTag = iConfig.getParameter<edm::InputTag>("inputMuonCollection");
-  m_muonsToken = consumes<reco::MuonCollection>(m_muonsTag);
-  m_candsTag = iConfig.getParameter<edm::InputTag>("inputCandCollection");
-  m_candsToken = consumes<reco::RecoChargedCandidateCollection>(m_candsTag);
-  m_previousCandTag = iConfig.getParameter<edm::InputTag>("previousCandTag");
-  m_previousCandToken = consumes<trigger::TriggerFilterObjectWithRefs>(m_previousCandTag);
-  m_minTrkHits = iConfig.getParameter<int>("minTrkHits");
-  m_minMuonHits = iConfig.getParameter<int>("minMuonHits");
-  m_minMuonStations = iConfig.getParameter<int>("minMuonStations");
-  m_maxNormalizedChi2 = iConfig.getParameter<double>("maxNormalizedChi2");
-  m_allowedTypeMask = iConfig.getParameter<unsigned int>("allowedTypeMask");
-  m_requiredTypeMask = iConfig.getParameter<unsigned int>("requiredTypeMask");
-  m_trkMuonId = muon::SelectionType(iConfig.getParameter<unsigned int>("trkMuonId"));
-  m_minPt = iConfig.getParameter<double>("minPt");
-  m_minN = iConfig.getParameter<unsigned int>("minN");
-  m_maxAbsEta = iConfig.getParameter<double>("maxAbsEta");
-}
+HLTMuonTrkFilter::HLTMuonTrkFilter(const edm::ParameterSet& iConfig)
+    : HLTFilter(iConfig),
+      propSetup_(iConfig, consumesCollector()),
+      m_muonsTag(iConfig.getParameter<edm::InputTag>("inputMuonCollection")),
+      m_muonsToken(consumes<reco::MuonCollection>(m_muonsTag)),
+      m_candsTag(iConfig.getParameter<edm::InputTag>("inputCandCollection")),
+      m_candsToken(consumes<reco::RecoChargedCandidateCollection>(m_candsTag)),
+      m_previousCandTag(iConfig.getParameter<edm::InputTag>("previousCandTag")),
+      m_previousCandToken(consumes<trigger::TriggerFilterObjectWithRefs>(m_previousCandTag)),
+      m_minTrkHits(iConfig.getParameter<int>("minTrkHits")),
+      m_minMuonHits(iConfig.getParameter<int>("minMuonHits")),
+      m_minMuonStations(iConfig.getParameter<int>("minMuonStations")),
+      m_maxNormalizedChi2(iConfig.getParameter<double>("maxNormalizedChi2")),
+      m_allowedTypeMask(iConfig.getParameter<unsigned int>("allowedTypeMask")),
+      m_requiredTypeMask(iConfig.getParameter<unsigned int>("requiredTypeMask")),
+      m_trkMuonId(muon::SelectionType(iConfig.getParameter<unsigned int>("trkMuonId"))),
+      m_minPt(iConfig.getParameter<double>("minPt")),
+      m_minN(iConfig.getParameter<unsigned int>("minN")),
+      m_maxAbsEta(iConfig.getParameter<double>("maxAbsEta")),
+      m_l1MatchingdR(iConfig.getParameter<double>("L1MatchingdR")),
+      m_l1MatchingdR2(m_l1MatchingdR * m_l1MatchingdR) {}
```

Comment:
```suggestion
      m_l1MatchingdR2(m_l1MatchingdR * m_l1MatchingdR) {
        if (m_l1MatchingdR <= 0.) {
          throw cms::Exception("HLTMuonTrkFilterConfiguration") << "invalid value for parameter \"L1MatchingdR\" (must be > 0): " << m_l1MatchingdR;
        }
      }
```
Please add an exception for the case where the DR config parameter is non-positive (as done in #37110). This comment also applies to other HLT filters in this PR.

Reply:
Done.